File: Coursework.API/Core/Entities/Origin/Match.cs
using Core.Models;
using System;

namespace Core.Entities.Origin
{
    class Match : IEntity
    {
        public enum Result
        {
            FirstTeamWon,
            SecondTeamWon,
            Tie,
            None
        }

        public string Id { get; set; }

        public DateTime Date { get; set; }
        public Result MatchResult { get; set; } = Result.None;

        public string GameId { get; set; }
        public Game Game { get; set; }

        public string TeamOneId { get; set; }
        public Team TeamOne { get; set; }

        public string TeamTwoId { get; set; }
        public Team TeamTwo { get; set; }
    }
}

File: coursework-front-end/src/app/models/match.ts
import { IModel } from './imodel';
import { MatchResult } from './match-result-enum';

export class Match implements IModel {
    constructor(
        public id?: string,
        public date?: Date,
        public result?: MatchResult,
        public gameId?: string,
        public teamOneId?: string,
        public teamTwoId?: string
    ) {}

    public static mock(): Match {
        return new Match("Mock", new Date(), MatchResult.Tie, "Mock game", "Mock t1", "Mock t2");
    }
}